### PR TITLE
#270: crash with weak viewController fixed

### LINKED ISSUE #270: Buddybuild crash report on build #474
Buddybuild detected a crash from raphael.koch@zalando.de

[View Full Crash Details](https://dashboard.buddybuild.com/apps/57a305cb34a9450100595b71/crashreport/54799ce649277c5878797aab04048584)

### PR DESCRIPTION
**Pull request by artdaw**
#270: during the checkout creation request a user might cancel Checkout Summary which causes self.viewController to be nil. That is why after the successful request we get `_swift_abortRetainUnowned` crash.

### CODE REVIEW

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift [hunk 1]
```diff
@@ -101,92 +108,99 @@ extension CheckoutSummaryActionsHandler {
...
+
         guard let paymentURL = order.externalPaymentUrl else {
-            self.viewController.viewState = .OrderPlaced
+            strongViewController.viewState = .OrderPlaced
             return
         }
 
-        let paymentSelectionViewController = PaymentSelectionViewController(paymentSelectionURL: paymentURL)
-        paymentSelectionViewController.paymentCompletion = { result in
+        let paymentSelectionstrongViewController = PaymentSelectionViewController(paymentSelectionURL: paymentURL)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Just `paymentSelectionViewController`

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift [hunk 2]
```diff
@@ -101,92 +108,99 @@ extension CheckoutSummaryActionsHandler {
...
     }
 
     internal func showBillingAddressSelectionScreen() {
+        guard let strongViewController = self.viewController else { return }
         guard Atlas.isUserLoggedIn() else { return loadCustomerData() }
-        let addressSelectionViewController = AddressPickerViewController(checkout: viewController.checkout, addressType: .billing,
-            addressSelectionCompletion: pickedAddressCompletion)
-        addressSelectionViewController.selectedAddress = viewController.checkoutViewModel.selectedBillingAddress
-        viewController.showViewController(addressSelectionViewController, sender: viewController)
+        let addressSelectionstrongViewController = AddressPickerViewController(checkout: strongViewController.checkout,
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Just `addressSelectionViewController`

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift [hunk 3]
```diff
@@ -101,92 +108,99 @@ extension CheckoutSummaryActionsHandler {
...
         }
-        viewController.showViewController(paymentSelectionViewController, sender: viewController)
+        strongViewController.showViewController(paymentSelectionstrongViewController, sender: strongViewController)
     }
 
     internal func showShippingAddressSelectionScreen() {
+        guard let strongViewController = self.viewController else { return }
         guard Atlas.isUserLoggedIn() else { return loadCustomerData() }
-        let addressSelectionViewController = AddressPickerViewController(checkout: viewController.checkout,
+        let addressSelectionstrongViewController = AddressPickerViewController(checkout: strongViewController.checkout,
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Just `addressSelectionViewController`

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift [hunk 4]
```diff
@@ -101,92 +108,99 @@ extension CheckoutSummaryActionsHandler {
...
 
     internal func showPaymentSelectionScreen() {
+        guard let strongViewController = self.viewController else { return }
         guard Atlas.isUserLoggedIn() else { return loadCustomerData() }
-        guard let paymentURL = viewController.checkoutViewModel.checkout?.payment.selectionPageUrl else { return }
+        guard let paymentURL = strongViewController.checkoutViewModel.checkout?.payment.selectionPageUrl else { return }
 
-        let paymentSelectionViewController = PaymentSelectionViewController(paymentSelectionURL: paymentURL)
-        paymentSelectionViewController.paymentCompletion = { result in
+        let paymentSelectionstrongViewController = PaymentSelectionViewController(paymentSelectionURL: paymentURL)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Just `paymentSelectionViewController`

File: AtlasUI/AtlasUITests/EditAddressTypeSpec.swift [hunk 5]
```diff
@@ -15,16 +15,16 @@ class EditAddressTypeSpec: QuickSpec {
 
         describe("Edit Address") {
             it("should format field title") {
-                expect(AddressFormField.Title.title(localizer)).to(equal("Title*"))
-                expect(AddressFormField.FirstName.title(localizer)).to(equal("First Name*"))
-                expect(AddressFormField.LastName.title(localizer)).to(equal("Last Name*"))
-                expect(AddressFormField.Street.title(localizer)).to(equal("Street*"))
+                expect(AddressFormField.Title.title(localizer)).to(equal("Title"))
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUITests/EditAddressTypeSpec.swift]: I have merged my changed again and the Title and all required fields should have `*`

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift (hunk 4 above)
Comment (artdaw) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Fixed

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift (hunk 3 above)
Comment (artdaw) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Fixed

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift (hunk 2 above)
Comment (artdaw) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Fixed

File: AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift (hunk 1 above)
Comment (artdaw) [inline AtlasUI/AtlasUI/CheckoutSummaryActionsHandler.swift]: Fixed

File: AtlasUI/AtlasUITests/EditAddressTypeSpec.swift (hunk 5 above)
Comment (artdaw) [inline AtlasUI/AtlasUITests/EditAddressTypeSpec.swift]: Test fixed

**Comment by bonkey** [comment]
👍

**Comment by artdaw** [comment]
👍

**Pull request merged**